sparse_ho/models/enet.py

```diff
@@ -118,6 +118,16 @@ def _get_pobj(r, X, beta, alphas, y=None):
         pobj += 0.5 * alphas[1] * norm(beta) ** 2
         return pobj
 
+    @staticmethod
+    def _get_dobj(r, X, beta, alpha, y=None):
+        # the dual variable is theta = (y - X beta) / (alpha[0] * n_samples)
+        n_samples = X.shape[0]
+        theta = r / (alpha[0] * n_samples)
+        dobj = alpha[0] * y @ theta
+        dobj -= alpha[0] ** 2 * n_samples / 2 * (theta ** 2).sum()
```

Comment:
```suggestion
        dobj -= alpha[0] ** 2 * n_samples / 2 * np.dot(theta, theta)
```

Reply:
This is more stable? Good to know :)

Reply:
no it avoids allocating an array contain the squares